Dockerfile

```diff
@@ -48,16 +51,41 @@ ENV CHPL_HOME=/opt/chapel \
     CHPL_GMP=system \
     CHPL_LLVM=system
 
-# acquire sources
 WORKDIR $CHPL_HOME
+
+
+# Build stage: build Chapel from sources
+# ======================================
+
+FROM chapel-base as chapel-build
+
+# acquire sources
 COPY . .
 
 # build Chapel for both C and LLVM backends
-RUN CHPL_TARGET_COMPILER=llvm make \
-    && CHPL_TARGET_COMPILER=gnu make \
-    && make chpldoc test-venv mason \
-    && make cleanall
+RUN CHPL_TARGET_COMPILER=llvm make -j10 \
+    && CHPL_TARGET_COMPILER=gnu make -j10
+RUN make chpldoc test-venv mason -j10
+RUN make chapel-py-venv chplcheck chpl-language-server -j10
+RUN make cleanall -j10
 
 # Hack to get access to Chapel binaries
 RUN cd $CHPL_HOME/bin && ln -s */* .
+
+# The .git folder is huge and we really don't need it.
+RUN rm -rf .git
+RUN for subdir in `ls test || true`; do \
```

Comment:
If you remove `.git` after this, you could just `rm -rf test` then `git restore test/release` if that's easier.

Reply:
Thanks, that's a good idea. However, I was informed that we don't include `test/release` in our release tarballs (and presumably we don't include `.git`, too?). As such, I think it's safer to explicitly perform a (possibly-empty) iteration on `test`.